modules/markup/external/external.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package external
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup"
+	"code.gitea.io/gitea/modules/setting"
+
+	gouuid "github.com/satori/go.uuid"
+)
+
+// RegisterParsers registers all supported third part parsers according settings
+func RegisterParsers() {
+	for _, render := range setting.ExternalMarkupRenders {
+		if render.Enabled && render.Command != "" && len(render.FileExtensions) > 0 {
```

Comment:
It might be good to validate the file extension to deny illegal file extensions.

Reply:
For example?

Reply:
`.<iframe>`

Reply:
But if you can change the ini file. I will trust you.

Reply:
We could only allow the regex \.\w ? This would prevent misconfiguration.

Reply:
@sapk Have checked that on https://github.com/go-gitea/gitea/pull/2570/files#diff-60f5fca6d29eafbb30eb645a0cd4041fR1074